fair/src/autogluon/fair/learners/fair.py

```diff
@@ -0,0 +1,621 @@
+            self.group_encoder.fit(self._internal_groups.reshape(-1, 1))
+            self._val_thresholds = self.group_encoder.transform(self._internal_groups.reshape(-1, 1)).toarray()
+
+        if callable(predictor):
+            self.proba = predictor(self.validation_data)
+        else:
+            self.proba = predictor.predict_proba(self.validation_data)
+        self.proba = np.asarray(self.proba)
+        self.y_true = np.asarray(validation_labels == self.predictor.class_labels[1])
+        self.frontier = None
+        if self.use_fast:
+            self.offset = np.zeros((self._val_thresholds.shape[1],))
+        else:
+            self.offset = np.zeros((self._val_thresholds.shape[1], self.proba.shape[1]))
+        self.objective1 = None
+        self.objective2 = None
+        self.round = False
+
+    def groups_to_numpy(self, groups, data: pd.DataFrame) -> np.ndarray:
+        """helper function for transforming groups into a numpy array of unique values
+        parameters
+        ----------
+        groups: one of the standard represenations of groups (see class doc)
+        data: a pandas dataframe
+        returns
+        -------
+        numpy array
+        """
+        if groups is None:
+            groups = self.groups
+        if callable(groups):
+            out = groups(data).argmax(1)
+        elif isinstance(groups, str):
+            out = np.asarray(data[groups])
+        elif groups is False:
+            out = np.zeros(data.shape[0])
+        else:
+            out = groups
+        return out
+
+    def fit(self, objective, constraint, value, greater_is_better_obj=None,
+            greater_is_better_const=None, recompute=True, round=False, budget=False):
+        """Fits the chosen predictor to optimize an objective while satisfing a constraint.
+        parameters
+        ----------
+        objective: a BaseGroupMetric or Scorable to be optimised
+        constraint: a BaseGroupMetric or Scorable that must be above/below a certain value
+        value: float the value constraint must be above or below
+        greater_is_better_obj: bool or None Governs if the objective is maximised (True) or  minimized (False).
+                If None the value of objective.greater_is_better is used.
+        greater_is_better_const: bool or None Governs if the constraint has to be greater (True) or  smaller (False) than value.
+                If None the value of constraint.greater_is_better is used.
+        recompute: governs if the the parato frontier should be recomputed. Use False to efficiently adjusting the threshold while
+                keeping objective and constraint fixed.
+        round: float or False. Can round the solutions found by predict_proba to prevent overfitting. Generally not needed.
+        returns
+        -------
+        Nothing
+        """
+        if greater_is_better_obj is None:
+            greater_is_better_obj = objective.greater_is_better
```

Comment:
Because it's not guaranteed that the object knows which way you want to optimize.
e.g. you might want to: 
maximize per group recall subject to the positive decision rate being < 0.5
or
maximize overall precision subject to the positive decision rate being > 0.5

So we have sane defaults and the ability to override.